### PR TITLE
feat: impl output dynfed log

### PR DESCRIPTION
**Pull request by kobayu858**
- impl output dynfed log
-- Schedulable or not
-- Basic information about dag_set, processor
-- Log information of dag_set, node_set, processor

### CODE REVIEW

File: 2021_RTCSA_dynfed/src/main.rs [hunk 1]
```diff
@@ -31,31 +37,83 @@ struct ArgParser {
     output_dir_path: String,
 }
 
+pub fn adjust_dag_set(dag_set: &mut [Graph<NodeData, i32>]) {
```
Comment (atsushi421) [inline 2021_RTCSA_dynfed/src/main.rs:40]: To clarify the process, how about the name `adjust_to_implicit_deadline` or `ensure_implicit_deadline`?
In addition, this process should be moved to util.rs and made a shared library.

File: 2021_RTCSA_dynfed/src/main.rs [hunk 2]
```diff
@@ -31,31 +37,83 @@ struct ArgParser {
...
+        adjust_dag(dag);
+    }
+}
+
+fn adjust_dag(dag: &mut Graph<NodeData, i32>) {
+    if let (Some(period), Some(end_to_end_deadline)) =
+        (dag.get_head_period(), dag.get_end_to_end_deadline())
+    {
+        handle_equal_period_and_deadline(dag, period, end_to_end_deadline);
+    } else if dag.get_head_period().is_none() {
```
Comment (atsushi421) [inline 2021_RTCSA_dynfed/src/main.rs:51]: It is recommended to store the result in a temporary variable, as it is inefficient to call `get_head_period` and `get_end_to_end_deadline` for every conditional statement.

File: 2021_RTCSA_dynfed/src/main.rs [hunk 3]
```diff
@@ -31,31 +37,83 @@ struct ArgParser {
...
+fn handle_missing_deadline(dag: &mut Graph<NodeData, i32>) {
+    let period = dag
+        .get_head_period()
+        .expect("Either an end-to-end deadline or period of time is required for the schedule.");
+    dag.add_param(
+        NodeIndex::new(dag.node_count() - 1),
+        "end_to_end_deadline",
+        period,
+    );
+}
```
Comment (atsushi421) [inline 2021_RTCSA_dynfed/src/main.rs:90]: The name of the function should be descriptive of what the function does.

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: 2021_RTCSA_dynfed/src/main.rs (hunk 1 above)
Comment (kobayu858) [reply 2021_RTCSA_dynfed/src/main.rs]: `adjust_dag_set` -> `adjust_to_implicit_deadline`
`main.rs` -> `util.rs`

File: 2021_RTCSA_dynfed/src/main.rs (hunk 2 above)
Comment (kobayu858) [reply 2021_RTCSA_dynfed/src/main.rs]: Refactoring has resulted in the use of match statements.

File: 2021_RTCSA_dynfed/src/main.rs (hunk 3 above)
Comment (kobayu858) [reply 2021_RTCSA_dynfed/src/main.rs]: Helper functions have been removed by refactoring.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Pull request merged**